### PR TITLE
44 cards component

### LINKED ISSUE #44: partners page
### Features to Add
- :heavy_check_mark: display all partners, along with connected ministers
- :heavy_check_mark: pull partners list from state
- :heavy_check_mark: will probably need to create a view table on db to get the connected ministers for each partner before sending initial request for partners
- :heavy_check_mark: add edit and delete functionality
- :heavy_check_mark: better formatting of address
- :heavy_check_mark: loading animations on edit and delete
- :heavy_check_mark: success/error alerts for edit and delete
- :heavy_check_mark: confirmation check before deleting
- :heavy_check_mark: paginate partners list
  - make sure each column is sorting in a way that makes sense - connected ministers should sort by number of connected ministers probably or not have the option to sort this column, and it needs to work on mobile and desktop
- :heavy_check_mark: sort partners list by names in alphabetical order before creating search index
- :heavy_check_mark: change birthdate to be two selection inputs?
- add form validation for edit
- have search only show exact matches based on names, email, minister names
- add functionality to set/change/remove spouse via partner edit form
  - :heavy_check_mark: add X button to remove existing spouse
  - allow arrow keys to move through searchList results
  - stop adding spouse name via postgres view table, just use id to get name from redux store
  - remove `is_married` column on partners, just check if `spouse_id` exists
  - use a merge table to connect spouses
- look into using [supabase realtime database subscriptions](https://supabase.com/docs/reference/javascript/subscribe)?

### Bugs
- :heavy_check_mark: may not be autofocusing when adding a name for partner
- :heavy_check_mark: when searching it's not automatically sorting by search result score

### PR DESCRIPTION
**Pull request by matthew-r-clark**

### CODE REVIEW

File: src/components/Card.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  right: 5,
+  borderRadius: '50%',
+  '&:hover': {
+    backgroundColor: colors.lightGray,
+  },
+});
+
+export default function Card({ children, isOpen, close }) {
+  return (
+    <Modal
```
Comment (MitchellPavel) [inline src/components/Card.js]: Is there a reason why you used styled for the Close Button and not the other components?

File: src/components/PartnerCard.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+              <AddressLine>
+                {`${partner.city}, ${partner.state} ${partner.zip_code}`}
+              </AddressLine>
+            </>
+          ) : 'n/a'}
+        </GridItem>
+
+        <GridItem xs={6}>
+          <H3>Birthday</H3>
+          {partner.birthday || 'n/a'}
```
Comment (MitchellPavel) [inline src/components/PartnerCard.js:58]: Do these need to be wrapped in HTML elements? Not sure if there would be any negative side effects of not doing it or if it matters.

Comment (matthew-r-clark) [reply src/components/PartnerCard.js:58]: Can't remember if I originally had these wrapped and if I had a reason or not, but I have an upcoming PR where I made some significant changes to the styling on this component.

File: src/components/Card.js (hunk 1 above)
Comment (matthew-r-clark) [reply src/components/Card.js]: iirc I was just adding styling inline and then used styled for that component so I could leverage the hover state. I went ahead and made changes to make this consistent.

**Pull request merged**